src/GafferScene/SceneAlgo.cpp

```diff
@@ -612,7 +612,11 @@ void addLocaliseAttributesPredecessors( const SceneAlgo::History::Predecessors &
 		}
 	}
 
-	assert( predecessor );
+	if( !predecessor )
+	{
+		return;
+	}
```

Comment:
I don't understand why this change is needed, since the idea was to keep all the changes in the Inspector class. Could you explain? Was the assert already wrong for some reason?

Reply:
Yes, that's what I understand it to be. I'm a bit nervous there's something lurking below the surface that this may conjure, but so far so good?

Reply:
All makes sense to me. Might be a tiny bit clearer to remove the assert in the same commit that added `Mode.Create` since that's the change that actually makes the assert invalid...